### PR TITLE
Add details of missing and stale values in rule testing docs

### LINKED ISSUE #7887: Unexpected behavior when using _ in alert unit test to generate a missing value
**What did you do?**

I used `_` in an alert unit test to generate a missing value

**What did you expect to see?**

An alert being triggered

**What did you see instead? Under which circumstances?**

The alert was not triggered. However, I found a workaround that shows this is unexpected behavior.

**Environment**

I have a repro that I based off `master` a few minutes ago: https://github.com/colltoaction/prometheus/commit/0d22fb1d7c550fad9bdc8552b67c92c42e098415. The files `cmd/promtool/testdata/unittest_missing_bad.yml` and `cmd/promtool/testdata/unittest_missing_good.yml` are almost identical, but one passes and the other one doesn't.

You can see that I only changed

```yaml
values: "0 _"
```

for

```yaml
values: "_ 0"
```

-----

I debugged the code as far as I could, but when I reached https://github.com/prometheus/prometheus/blob/267dd17fe7c847fecb5255dc88278f5f1f8ee92b/promql/parser/parse.go#L200 I was basically lost. It reached the point where the code was using a yacc generated parser and I didn't want to dig further before posting here.

### PR DESCRIPTION
**Pull request by dgl**
Fixes #7887 (along with e7e6062 which added some unit tests using _ and
stale).

Signed-off-by: David Leadbeater <dgl@dgl.cx>

### CODE REVIEW

File: docs/configuration/unit_testing_rules.md [hunk 1]
```diff
@@ -72,12 +72,23 @@ series: <string>
 # Expanding notation:
 #     'a+bxc' becomes 'a a+b a+(2*b) a+(3*b) … a+(c*b)'
 #     'a-bxc' becomes 'a a-b a-(2*b) a-(3*b) … a-(c*b)'
+#     '  axc' becomes 'a a … a'
 # Examples:
 #     1. '-2+4x3' becomes '-2 2 6 10'
 #     2. ' 1-2x4' becomes '1 -1 -3 -5 -7'
+#     3. '   1x4' becomes '1 1 1 1'
```
Comment (brian-brazil) [inline docs/configuration/unit_testing_rules.md:79]: This simple example should probably go first

File: docs/configuration/unit_testing_rules.md [hunk 2]
```diff
@@ -72,12 +72,23 @@ series: <string>
...
+#
+# '_' may be used in place of a number to indicate no value at that time.
+#
+# Additionally the keyboard 'stale' may be used to explicitly mark a series as
```
Comment (brian-brazil) [inline docs/configuration/unit_testing_rules.md:83]: Not quite, it's a stale marker at that timestamp which isn't the same as marking an entire series stale (which you can't do).

File: docs/configuration/unit_testing_rules.md [hunk 3]
```diff
@@ -72,12 +72,23 @@ series: <string>
...
+# stale.
 values: <string>
 ```
 
+`stale` is not possible to repeat using the expanding notation, but it is not
```
Comment (brian-brazil) [inline docs/configuration/unit_testing_rules.md:88]: This is not how staleness work, this would only confuse matters. I think it'd be best to remote this.

**Comment by roidelapluie** [comment]
Hello @dgl , can we move on here? thanks!

**Comment by ianbillett** [comment]
This issue has come up in the mailing list [here](https://groups.google.com/g/prometheus-users/c/j2F1rgQ6YoA/m/GWJMlXHUBAAJ) - it would be good to get this merged :rocket:

**Comment by dgl** [comment]
Fixed by #9065

**Pull request closed without merging**